node/rpc.go

```diff
@@ -0,0 +1,150 @@
+package node
+
+import (
+	"context"
+	"errors"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/rs/cors"
+	cfg "github.com/tendermint/tendermint/config"
+	"github.com/tendermint/tendermint/libs/log"
+	tmpubsub "github.com/tendermint/tendermint/libs/pubsub"
+	"github.com/tendermint/tendermint/libs/strings"
+	rpccore "github.com/tendermint/tendermint/rpc/core"
+	grpccore "github.com/tendermint/tendermint/rpc/grpc"
+	rpcserver "github.com/tendermint/tendermint/rpc/jsonrpc/server"
+	"github.com/tendermint/tendermint/types"
+)
+
+func startGRPCServer(rpcConfig *cfg.RPCConfig,
```

Comment:
Optional cosmetic nit: 

I would like to advocate against folding function parameters like this. I see we often do it, but it makes it harder to read and to grep for them. It's OK, and at least in my opinion preferable, for the line to be a little long. 

```go
func startGRPCServer(config *cfg.RPCConfig, env *rpccore.Environment, logger log.Logger) (net.Listener, error) {
```

This also has the advantage that if the parameter list is so long that it's _hard_ to read this way, you feel the pinch before the code lands in the repo rather than later. 🙂

Reply:
On a related note: I recommend _not_ echoing type information in the names of variables and parameters. The type declaration isn't far away, if it matters—and ideally the name should be semantic rather than structural.

Reply:
@creachadair  Mind clarifying what you mean by the distinction between structural vs semantic here?

Reply:
I only mean: the difference between "what it is" (the structural type of the value) and "what it's for" (the intended use of the value). Since the structural type `RPCConfig` is already stated, it needn't be repeated in the name of the variable (IMO), since what matters to the function is that it is the operative configuration this function is using.

(Personally I'd have just written `cfg` here, but I know not everyone is as happy with abbreviation as I am)